### PR TITLE
DWPF-1635 Tagged PageList

### PR DESCRIPTION
**Pull request by 0RWilliams**
Adds a new Tagged Pages component, that generates a list of links based on a single tag.
Currently this list is unlimited, and returns all live pages tagged.

- [x] JIRA ticket referenced in title
- [x] Title is clear and concise
- [x] Description gives any relevant detail
- [x] Tests are up to date
- [x] Documentation is up to date

### CODE REVIEW

**Comment by 0RWilliams** [comment]
Example of Admin:
image

**Comment by 0RWilliams** [comment]
Nav page example:
image

File: src/content/models.py [hunk 1]
```diff
@@ -450,6 +450,7 @@ class NavigationPage(BasePage):
     secondary_elements = StreamField(
         [
             ("dw_curated_page_links", dwds_blocks.CustomPageLinkListBlock()),
+            ("dw_tagged_page_List", dwds_blocks.TaggedPageListBlock()),
```
Comment (marcelkornblum) [inline src/content/models.py:453]:
```suggestion
            ("dw_tagged_page_list", dwds_blocks.TaggedPageListBlock()),
```

File: src/dw_design_system/dwds/components/link_list.py [hunk 2]
```diff
@@ -1,5 +1,71 @@
...
+
+class TaggedPageListBlock(blocks.StructBlock):
+    title = blocks.CharBlock(
+        label="Title",
+        icon="title",
+        classname="full title",
+        max_length=40,
+        search_index=False,
+    )
+    description = blocks.CharBlock(required=False, max_length=40)
```
Comment (marcelkornblum) [inline src/dw_design_system/dwds/components/link_list.py:27]: can you confirm the lengths for title and description match the other components please? I think they might be 25 and 50 ?

File: src/dw_design_system/dwds/components/link_list.py [hunk 3]
```diff
@@ -1,5 +1,71 @@
...
+    tag = blocks.ChoiceBlock(label="Tag", choices=get_tag_choices, required=True)
+    ordering_choice = blocks.ChoiceBlock(
```
Comment (marcelkornblum) [inline src/dw_design_system/dwds/components/link_list.py:29]: <3

File: src/dw_design_system/dwds/components/link_list.py [hunk 4]
```diff
@@ -1,5 +1,71 @@
...
+    def get_context(self, value, parent_context=None):
+        ordering = value["ordering_choice"] or "-content_object__last_published_at"
+        context = parent_context or {}
+
+        # Query tag via slug, return nothing if not found
+        tag = Tag._default_manager.all().get(slug=value["tag"])
+        if tag:
+            pages = (
+                TaggedPage.objects.filter(tag=tag)
+                .filter(content_object__live=True)
```
Comment (marcelkornblum) [inline src/dw_design_system/dwds/components/link_list.py:51]: do we need `public` as a filter in here as well as live?

File: src/dw_design_system/dwds/components/link_list.py [hunk 5]
```diff
@@ -1,5 +1,71 @@
...
+        icon = "tag"
+
+    def get_context(self, value, parent_context=None):
+        ordering = value["ordering_choice"] or "-content_object__last_published_at"
+        context = parent_context or {}
+
+        # Query tag via slug, return nothing if not found
+        tag = Tag._default_manager.all().get(slug=value["tag"])
+        if tag:
+            pages = (
```
Comment (marcelkornblum) [inline src/dw_design_system/dwds/components/link_list.py:49]: this is a List in the else section - let's keep the type consistent across both

**Review by marcelkornblum — COMMENTED** [review]
nice work!

File: src/dw_design_system/dwds/components/link_list.py (hunk 2 above)
Comment (alsyx) [reply src/dw_design_system/dwds/components/link_list.py:27]: I think for title it is 30 and description is 70.

File: src/dw_design_system/dwds/components/link_list.py (hunk 5 above)
Comment (0RWilliams) [reply src/dw_design_system/dwds/components/link_list.py:49]: replaced.

File: src/dw_design_system/dwds/components/link_list.py (hunk 4 above)
Comment (0RWilliams) [reply src/dw_design_system/dwds/components/link_list.py:51]: added public as well.

File: src/content/models.py (hunk 1 above)
Comment (0RWilliams) [reply src/content/models.py:453]: Done.

File: src/dw_design_system/dwds/components/link_list.py (hunk 2 above)
Comment (0RWilliams) [reply src/dw_design_system/dwds/components/link_list.py:27]: Changed to 30 & 70.

File: src/dw_design_system/dwds/components/link_list.py (hunk 5 above)
Comment (0RWilliams) [reply src/dw_design_system/dwds/components/link_list.py:49]: @marcelkornblum , is this better?

**Review by marcelkornblum — APPROVED** [review]

**Pull request merged**